src/Symfony/Component/HttpFoundation/Tests/Session/Storage/Handler/MongoDbSessionHandlerTest.php

```diff
@@ -11,56 +11,98 @@
+use MongoDB\Driver\BulkWrite;
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\CommandException;
+use MongoDB\Driver\Exception\ConnectionException;
+use MongoDB\Driver\Manager;
+use MongoDB\Driver\Query;
 use PHPUnit\Framework\TestCase;
 use Symfony\Component\HttpFoundation\Session\Storage\Handler\MongoDbSessionHandler;
 
+require_once __DIR__.'/stubs/mongodb.php';
+
 /**
  * @author Markus Bachmann <markus.bachmann@bachi.biz>
  *
+ * @group integration
  * @group time-sensitive
  *
  * @requires extension mongodb
  */
 class MongoDbSessionHandlerTest extends TestCase
 {
+    private const DABASE_NAME = 'sf-test';
+    private const COLLECTION_NAME = 'session-test';
+
     public array $options;
-    private MockObject&Client $mongo;
+    private Manager $manager;
     private MongoDbSessionHandler $storage;
 
-    public static function setUpBeforeClass(): void
-    {
-        if (!class_exists(Client::class)) {
-            throw new SkippedTestSuiteError('The mongodb/mongodb package is required.');
-        }
-    }
-
     protected function setUp(): void
     {
         parent::setUp();
 
-        $this->mongo = $this->getMockBuilder(Client::class)
-            ->disableOriginalConstructor()
-            ->getMock();
+        $this->manager = new Manager('mongodb://'.getenv('MONGODB_HOST'));
+
+        try {
+            $this->manager->executeCommand(self::DABASE_NAME, new Command(['ping' => 1]));
+        } catch (ConnectionException $e) {
+            $this->markTestSkipped(sprintf('MongoDB Server "%s" not running: %s', getenv('MONGODB_HOST'), $e->getMessage()));
+        }
 
         $this->options = [
             'id_field' => '_id',
             'data_field' => 'data',
             'time_field' => 'time',
             'expiry_field' => 'expires_at',
-            'database' => 'sf-test',
-            'collection' => 'session-test',
+            'database' => self::DABASE_NAME,
+            'collection' => self::COLLECTION_NAME,
         ];
 
-        $this->storage = new MongoDbSessionHandler($this->mongo, $this->options);
+        $this->storage = new MongoDbSessionHandler($this->manager, $this->options);
+    }
+
+    public function testCreateFromClient()
+    {
+        $client = $this->createMock(Client::class);
+        $client->expects($this->once())
+            ->method('getManager')
+            ->willReturn($this->manager);
+
+        $this->storage = new MongoDbSessionHandler($client, $this->options);
+        $this->storage->write('foo', 'bar');
+
+        $this->assertCount(1, $this->getSessions());
     }
 
-    public function testConstructorShouldThrowExceptionForMissingOptions()
+    protected function tearDown(): void
+    {
+        try {
+            $this->manager->executeCommand(self::DABASE_NAME, new Command(['drop' => self::COLLECTION_NAME]));
+        } catch (CommandException $e) {
+            // The server may return an error if the collection does not exist
```

Comment:
```suggestion
            // The server may return a NamespaceNotFound error if the collection does not exist
```

If we're not creating a constant for `26` below, I think it's a good idea to note the error name here for reference.

Reply:
Fixed.